cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -735,6 +737,69 @@ TEST_F(TestWriteRecordBatch, WriteWithCompression) {
   }
 }
 
+TEST_F(TestWriteRecordBatch, WriteWithCompressionAndMinSavings) {
+  // A small batch that's known to be compressible
+  auto batch = RecordBatchFromJSON(schema({field("n", int64())}), R"([
+    {"n":0},{"n":1},{"n":2},{"n":3},{"n":4},
+    {"n":5},{"n":6},{"n":7},{"n":8},{"n":9}])");
+
+  auto prefixed_size = [](const Buffer& buffer) -> int64_t {
+    if (buffer.size() < int64_t(sizeof(int64_t))) return 0;
+    return bit_util::FromLittleEndian(util::SafeLoadAs<int64_t>(buffer.data()));
+  };
+  auto content_size = [](const Buffer& buffer) -> int64_t {
+    return buffer.size() - sizeof(int64_t);
```

Comment:
nit: should this have a check like prefixed_size?

Reply:
It should be validated by the check in the loop (`ASSERT_GT(compressed_size, 0)`).

Main reason for the check in prefixed_size was because the memory-read could theoretically yield a positive value even if `buffer.size() < 8` (due to padding).